src/main/java/com/ciandt/techgallery/service/impl/GooglePlusCommunicationServiceImpl.java

```diff
@@ -0,0 +1,195 @@
+package com.ciandt.techgallery.service.impl;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.servlet.http.HttpServletRequest;
+
+import com.ciandt.techgallery.persistence.model.TechGalleryUser;
+import com.ciandt.techgallery.service.GooglePlusCommunicationService;
+import com.ciandt.techgallery.service.UserServiceTG;
+import com.ciandt.techgallery.service.enums.FeatureEnum;
+import com.ciandt.techgallery.service.enums.ValidationMessageEnums;
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.server.spi.response.BadRequestException;
+import com.google.api.server.spi.response.InternalServerErrorException;
+import com.google.api.server.spi.response.NotFoundException;
+import com.google.api.services.plusDomains.PlusDomains;
+import com.google.api.services.plusDomains.model.Acl;
+import com.google.api.services.plusDomains.model.Activity;
+import com.google.api.services.plusDomains.model.PlusDomainsAclentryResource;
+import com.google.appengine.api.users.User;
+
+/**
+ * Services for Google Plus communications Endpoint requests.
+ *
+ * @author Thulio Ribeiro
+ *
+ */
+public class GooglePlusCommunicationServiceImpl implements GooglePlusCommunicationService {
+
+  private static GooglePlusCommunicationServiceImpl instance;
+
+  private static final String POSITIVE_RECOMMENDATION_TEXT = "positivamente a tecnologia ";
+  private static final String NEGATIVE_RECOMMENDATION_TEXT = "negativamente a tecnologia ";
+  private UserServiceTG userService = UserServiceTGImpl.getInstance();
+  public static final String NEW_LINE = System.getProperty("line.separator");
+
+  private GooglePlusCommunicationServiceImpl() {
+  }
+
+  /**
+   * Singleton method for the service.
+   *
+   * @author <a href="mailto:tribeiro@ciandt.com"> Thulio Soares Ribeiro </a>
+   * @since 23/10/2015
+   *
+   * @return GooglePlusCommunicationServiceImpl instance.
+   */
+  public static GooglePlusCommunicationServiceImpl getInstance() {
+    if (instance == null) {
+      instance = new GooglePlusCommunicationServiceImpl();
+    }
+    return instance;
+  }
+
+  /*
+   * (non-Javadoc)
+   * 
+   * @see com.ciandt.techgallery.service.GooglePlusCommunicationService#
+   * postGooglePlus(com.ciandt.techgallery.service.enums.FeatureEnum,
+   * java.lang.Boolean, java.lang.String, java.lang.String, java.lang.String,
+   * com.google.appengine.api.users.User, javax.servlet.http.HttpServletRequest)
+   */
+  @Override
+  public void postGooglePlus(FeatureEnum feature, Boolean score, String comment, String currentUserMail,
```

Comment:
@tsr-thulio A service class should not know about HttpRequests. An endpoint should pass the AccessToken to the Service class.